src/cid-util.js

```diff
@@ -0,0 +1,36 @@
+'use strict'
+
+const mh = require('multihashes')
+
+var CIDUtil = {
+  /**
+   * Test if the given input is a valid CID object.
+   * Returns an error message if it is not, or
+   * Throws an error its multihash is invalid.
```

Comment:
Let's simplify it and just make it throw an error if it is doesn't have valid CID components, the error message will tell which component is invalid.

Reply:
@diasdavid I'm a little bit confused. 

The new implementation of `validateCID()` already does that. And, this behavior has been preserved from the existing implementation. The purpose of `checkCIDComponents()` was to use the same logic in `isCID()` without relying on the error being thrown. The idea was that `isCID()` was slow because of the error objects being created on the heap (which obviously we need to verify in the long run). If we change `checkCIDComponents()` to throw errors for all these conditions, we are effectively going back to the old implementation where `isCID()` directly uses `validateCID()`. The purpose of `checkCIDComponents()` was to avoid using error objects in CID validation in the constructor (via `isCID()`).

Please advise.

Reply:
Oh, I see.

If that is the case. Let's wrap the mh.validate on a tryCatch block inside the checkCIDComponents itself and only make that function return strings or undefined (should enable V8 to optimize it even more).

Reply:
OK. I'll check in the updated changes shortly.